das/redundant_simple_das_reader.go

```diff
@@ -64,3 +65,42 @@ func (r RedundantSimpleDASReader) HealthCheck(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (r *RedundantSimpleDASReader) ExpirationPolicy(ctx context.Context) (arbstate.ExpirationPolicy, error) {
+	// If at least one inner service has KeepForever,
+	// then whole redundant service can serve after timeout.
+	for _, serv := range r.inners {
+		expirationPolicy, err := serv.ExpirationPolicy(ctx)
+		if err != nil {
+			return -1, err
+		}
+		if expirationPolicy == arbstate.KeepForever {
+			return arbstate.KeepForever, nil
+		}
+	}
+	// If no inner service has KeepForever,
+	// but at least one inner service has DiscardAfterArchiveTimeout,
+	// then whole redundant service can serve till archive timeout.
+	for _, serv := range r.inners {
+		expirationPolicy, err := serv.ExpirationPolicy(ctx)
+		if err != nil {
+			return -1, err
+		}
+		if expirationPolicy == arbstate.DiscardAfterArchiveTimeout {
+			return arbstate.DiscardAfterArchiveTimeout, nil
+		}
+	}
+	// If no inner service has KeepForever, DiscardAfterArchiveTimeout,
+	// but at least one inner service has DiscardAfterDataTimeout,
+	// then whole redundant service can serve till data timeout.
+	for _, serv := range r.inners {
+		expirationPolicy, err := serv.ExpirationPolicy(ctx)
+		if err != nil {
+			return -1, err
+		}
+		if expirationPolicy == arbstate.DiscardAfterDataTimeout {
+			return arbstate.DiscardAfterDataTimeout, nil
+		}
+	}
+	return -1, errors.New("unknown expiration policy")
```

Comment:
I wonder if there's a way to do this with `math.Min/Max` given that it's an integer enum.